### PR TITLE
Maven friendly java project

### PR DESCRIPTION
**Pull request by srnagar**

### CODE REVIEW

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class DefaultAzureServiceBusCredentialTest {
+
+    /**
+     * Unit test to use {@link DefaultAzureCredential} to create a {@link TopicClient} and send a message.
+     * @throws ServiceBusException
+     * @throws InterruptedException
+     */
+    @Test
+    public void testDefaultCredential() throws ServiceBusException, InterruptedException {
+        TopicClient topicClient = new TopicClient("jongsb2", "srnagartopic",
```
Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Please implement the Environment variable scheme from .NET version, read in the .env file that is in the root.  Thanks.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            new ClientSettings(new DefaultAzureServiceBusCredential()));
+        topicClient.send(new Message("hello")); // if this fails, exception will be thrown failing the test
```
Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: See the .NET test, you are going to want to write to the topic and then read using a subscription, so we know that it works.

Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Do we have async versions?   I like to keep tests async too.

File: java/src/main/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredential.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  public DefaultAzureServiceBusCredential(DefaultAzureCredential defaultAzureCredential) {
+    this.defaultAzureCredential = defaultAzureCredential;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public CompletableFuture<SecurityToken> getSecurityTokenAsync(String audience) {
+    TokenRequestContext tokenRequestContext = new TokenRequestContext().addScopes(SERVICEBUS_SCOPE);
```
Comment (jongio) [inline java/src/main/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredential.java]: Once you implement the others Mgmt and Fluent, you'll see how I cache in-memory.  Can you also implement that here?  I think once you implement those other two flavors you'll get a better idea of what can be shared.

Comment (srnagar) [reply java/src/main/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredential.java]: Added caching

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java (hunk 1 above)
Comment (srnagar) [reply java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Reading from environment file in classpath

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java (hunk 2 above)
Comment (srnagar) [reply java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Added subscription to read but couldn't test as my VM didn't have Service Bus receive permissions.

Comment (srnagar) [reply java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Yes, using async version now.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ */
+public class DefaultAzureServiceBusCredentialTest {
+
+    private static final String AZURE_SERVICE_BUS_ENDPOINT = "AZURE_SERVICE_BUS_ENDPOINT";
+    private static final String AZURE_SERVICE_BUS_ENTITY_PATH = "AZURE_SERVICE_BUS_ENTITY_PATH";
+    private static Properties PROPERTIES;
+
+    @BeforeAll
+    static void loadEnvironmentProperties() {
+        try (InputStream inputStream = DefaultAzureServiceBusCredentialTest.class.getClassLoader().getResourceAsStream(".env.temp")) {
```
Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: You will want to use .env here, not .env.temp

Copy .env.temp, rename it to .env, put your secrets in there, and then run tests.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ * Unit test for {@link DefaultAzureServiceBusCredential}.
+ */
+public class DefaultAzureServiceBusCredentialTest {
+
+    private static final String AZURE_SERVICE_BUS_ENDPOINT = "AZURE_SERVICE_BUS_ENDPOINT";
+    private static final String AZURE_SERVICE_BUS_ENTITY_PATH = "AZURE_SERVICE_BUS_ENTITY_PATH";
+    private static Properties PROPERTIES;
+
+    @BeforeAll
+    static void loadEnvironmentProperties() {
```
Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Please use a lib like this: https://github.com/cdimascio/java-dotenv

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+     * @throws ServiceBusException
+     * @throws InterruptedException
+     */
+    @Test
+    public void testDefaultCredential() throws Exception {
+        ClientSettings clientSettings = new ClientSettings(new DefaultAzureServiceBusCredential());
+        TopicClient topicClient = new TopicClient(PROPERTIES.getProperty(AZURE_SERVICE_BUS_ENDPOINT),
+            PROPERTIES.getProperty(AZURE_SERVICE_BUS_ENTITY_PATH),
+            clientSettings);
+        String message = "hello";
```
Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: You'll want the message to be unique per test run. That way when you receive the message you know it is the one you just sent. 

Please see the .NET test on how it is done there.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+        // send a message
+        topicClient.sendAsync(new Message(message)).get();
+        topicClient.closeAsync().get();
+
+        // receive the message
+        SubscriptionClient subscriptionClient = new SubscriptionClient(PROPERTIES.getProperty(AZURE_SERVICE_BUS_ENDPOINT),
+            PROPERTIES.getProperty(AZURE_SERVICE_BUS_ENDPOINT), clientSettings, ReceiveMode.RECEIVEANDDELETE);
+        subscriptionClient.registerMessageHandler(new IMessageHandler() {
```
Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Seems like we could use a line break here between the ctor and the registerMessageHandler call.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+                return CompletableFuture.completedFuture(null);
+            }
+
+            @Override
+            public void notifyException(Throwable throwable, ExceptionPhase exceptionPhase) {
+
+            }
+        }, Executors.newCachedThreadPool());
+        subscriptionClient.closeAsync().get();
+
```
Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Remove extra lines.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit test for {@link DefaultAzureServiceBusCredential}.
+ */
+public class DefaultAzureServiceBusCredentialTest {
+
+    private static final String AZURE_SERVICE_BUS_ENDPOINT = "AZURE_SERVICE_BUS_ENDPOINT";
```
Comment (jongio) [inline java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: FYI - You are also going to need to set the other env vars to test envcred, but you don't have to do that incode, just in the .env and the dotenv package will pick it up, add it to env vars, and then envcred will pick them up.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java (hunk 2 above)
Comment (jongio) [reply java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Okay, please assign those perms and test.  Want to make sure it works.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java (hunk 9 above)
Comment (srnagar) [reply java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Tested with all environment variables locally and everything worked fine (not checking in the service bus names, client ids etc)

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java (hunk 4 above)
Comment (srnagar) [reply java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Changed the file name to `.env` and added javadoc on the usage.

File: java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java (hunk 5 above)
Comment (srnagar) [reply java/src/test/java/com/jongallant/azure/identity/extensions/DefaultAzureServiceBusCredentialTest.java]: Using the dependency now.

**Pull request merged**